source/rules/literals.rs

```diff
@@ -331,7 +331,7 @@ fn hexadecimal_mapper(span: Span) -> StdResult<Literal, ParseIntError> {
         )
         .and_then(
             |hexadecimal| {
-                Ok(Literal::Integer(Token::new(hexadecimal, span)))
+                Ok(Literal::Real(Token::new(hexadecimal as f64, span)))
```

Comment:
I understand what you did here. Excuse me if it was not clear in #44, but the goal is that it must _overflow_ to a real.

When the hexadecimal value does not fit in `i64` (`Literal::Integer`), it must be then be a `f64` (`Literal::Real`).

Moving everything to real is not optimized, and can complicate things later.

Reply:
Aaaah ok, I understand, more stuff like decimal rules ? 
```rust
i64::from_str(string)
        .and_then(...)
        .or_else(
            |_: ParseIntError| {
                f64::from_str(string).and_then(...)
            }
```

Reply:
Correct. But that's not trivial. In Zend Engine, this conversion from hexa to real is wrong most of the time, and the specifation says nothing about that. So that's not a trivial things to do. Maybe there is a crate on crates.io to help us.

Reply:
I forget that => bug-compliance that's wright ? Maybe I delete this PR cause more complicated than expected

Reply:
I haven't decided yet if we should be bug-compliant here or not. I guess not, and then I would have to fix Zend Engine. That's the best we can do. But until that, yes, if this task is too difficult, feel free to close this PR :-). Would you like another one?

Reply:
Yes of course :)

Reply:
https://github.com/tagua-vm/parser/issues/92 this one might be easy :-)